src/products/components/ProductVariantPage/ProductVariantPage.tsx

```diff
@@ -223,175 +223,190 @@ const ProductVariantPage: React.FC<ProductVariantPageProps> = ({
             handlers,
             hasChanged,
             submit
-          }) => (
-            <>
-              <Grid variant="inverted">
-                <div>
-                  <ProductVariantNavigation
-                    current={variant ? variant.id : undefined}
-                    defaultVariantId={defaultVariantId}
-                    fallbackThumbnail={maybe(
-                      () => variant.product.thumbnail.url
-                    )}
-                    variants={maybe(() => variant.product.variants)}
-                    onAdd={onAdd}
-                    onRowClick={(variantId: string) => {
-                      if (variant) {
-                        return onVariantClick(variantId);
-                      }
-                    }}
-                    onReorder={onVariantReorder}
-                  />
-                </div>
-                <div>
-                  <VariantDetailsChannelsAvailabilityCard variant={variant} />
-                  <Attributes
-                    entityId={variant?.id}
-                    title={intl.formatMessage(messages.nonSelectionAttributes)}
-                    attributes={data.attributes.filter(
-                      attribute =>
-                        attribute.data.variantAttributeScope ===
-                        VariantAttributeScope.NOT_VARIANT_SELECTION
-                    )}
-                    attributeValues={attributeValues}
-                    loading={loading}
-                    disabled={loading}
-                    errors={errors}
-                    onChange={handlers.selectAttribute}
-                    onMultiChange={handlers.selectAttributeMultiple}
-                    onFileChange={handlers.selectAttributeFile}
-                    onReferencesRemove={handlers.selectAttributeReference}
-                    onReferencesAddClick={onAssignReferencesClick}
-                    onReferencesReorder={handlers.reorderAttributeValue}
-                    fetchAttributeValues={fetchAttributeValues}
-                    fetchMoreAttributeValues={fetchMoreAttributeValues}
-                    onAttributeSelectBlur={onAttributeSelectBlur}
-                  />
-                  <CardSpacer />
-                  <Attributes
-                    entityId={variant?.id}
-                    title={intl.formatMessage(
-                      messages.selectionAttributesHeader
-                    )}
-                    attributes={data.attributes.filter(
-                      attribute =>
-                        attribute.data.variantAttributeScope ===
-                        VariantAttributeScope.VARIANT_SELECTION
+          }) => {
+            const nonSelectionAttributes = data.attributes.filter(
```

Comment:
`extractVariantSelectionAttributes` and `extractNotVariantSelectionAttributtes` plz

Reply:
Why? It's simple and non-reusable filter. Proposed function's name is almost as long as function body.

Reply:
Simply for the sake of readability. I think as a general rule if we need to extract consts from the places they're being used in the jsx then putting them somehwere else in the jsx makes it less readable and clean than if we placed them outside of return

Reply:
Its readability gain is close to none, but have it your way